pkg/sensors/tracing/generickprobe.go

```diff
@@ -954,7 +974,19 @@ func handleGenericKprobe(r *bytes.Reader) ([]observer.Event, error) {
 			}
 
 			arg.Index = uint64(a.index)
-			arg.Value = handleGenericKprobeString(r)
+			arg.Value, err = parseString(r)
+			if err != nil {
+				if errors.Is(err, errParseStringSize) {
+					// If no size then path walk was not possible and file was
+					// either a mount point or not a "file" at all which can
+					// happen if running without any filters and kernel opens an
+					// anonymous inode. For this lets just report its on "/" all
+					// though pid filtering will mostly catch this.
```

Comment:
I'm not sure I get this comment so maybe we should not apply this for `gt.GenericFileType, gt.GenericFdType, gt.GenericKiocb:` and `case gt.GenericPathType:`. I just reestablished it to have exactly the same behavior.

Reply:
It looks like the case I'm trying to handle is the anonymous file case where the value returned here is empty. I guess its fine to do for all the other cases as well. What we need to be sure is we report something so the logs at least have an entry.

Reply:
ok I see, I just have one last question, in the patched that introduced that you wrote:
```text
When using fd_install hook to monitor program open* if the open is against
the mount then our path walking logic will return the size=0. The user
side then interprets this as an error and prints an ugly warning.
```
And the thing I wonder is, if the size is indeed 0 and it's correctly writing a 4 bytes 0, decoding should be okay and not error:
https://github.com/cilium/tetragon/blob/b2e3858a05da4305bbf2b97d999869b87a2c83a1/bpf/process/types/basic.h#L412
But I guess you encountered the error? Because the only situation I see that could trigger this would be that BPF sends a message less than 4 bytes in total for that arg, thus we would get an EOF.

Reply:
Ok maybe this is never triggered anymore